lightning/src/chain/onchaintx.rs

```diff
@@ -404,10 +426,26 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 			preprocessed_requests.push(req);
 		}
 
+		loop {
+			let pop_height = if let Some(first_entry) = self.locktimed_packages.iter().next() {
+				if *first_entry.0 <= height + 1 {
+					*first_entry.0
+				} else { break; }
+			} else { break; };
+			log_trace!(logger, "Restoring delayed claim of package(s) at their timelock at {}.", pop_height);
+			preprocessed_requests.append(&mut self.locktimed_packages.remove(&pop_height).unwrap());
+		}
+
 		// Generate claim transactions and track them to bump if necessary at
 		// height timer expiration (i.e in how many blocks we're going to take action).
 		for mut req in preprocessed_requests {
-			if let Some((new_timer, new_feerate, tx)) = self.generate_claim_tx(height, &req, &*fee_estimator, &*logger) {
+			if req.package_timelock() > height + 1 {
+				log_debug!(logger, "Delaying claim of package until its timelock at {} (current height {}), the following outpoints are spent:", req.package_timelock(), height);
+				for outpoint in req.outpoints() {
+					log_debug!(logger, "  Outpoint {}", outpoint);
+				}
+				self.locktimed_packages.entry(req.package_timelock()).or_insert(Vec::new()).push(req);
```

Comment:
I think I would gather all preprocessing checks (aggregation, delayed-claiming) in the same for loop above and the restoring loop as the first thing in `update_claims_view`. Post-anchor, we might be interested to aggregate our HTLC-timeout with an equal cltv_expiry and it won't work with this code organization.